### PR TITLE
452: Implement solar tech info text into form

### PR DESCRIPTION
**Pull request by daren-thomas**
This PR implements #452. To test, open the energy systems form and select either PV or ST. Changing the module type will change the text displayed at the bottom left.

### CODE REVIEW

File: src/Hive.IO/Hive.IO/Forms/ConversionTechPropertiesViewModel.cs [hunk 1]
```diff
@@ -175,6 +175,8 @@ public ModuleTypeRecord ModuleType
             }
         }
 
+        public string Description => IsSurfaceTech ? ModuleType.Description : Defaults[Name].Description;
```
Comment (christophwaibel) [inline src/Hive.IO/Hive.IO/Forms/ConversionTechPropertiesViewModel.cs:178]: so do I understand correctly, if it's not a surfaceTech, `description` will simply be not instantiated?

Comment (daren-thomas) [reply src/Hive.IO/Hive.IO/Forms/ConversionTechPropertiesViewModel.cs:178]: If it's not a surface tech, then the description is taken from the `Defaults[Name]` object.

Comment (christophwaibel) [reply src/Hive.IO/Hive.IO/Forms/ConversionTechPropertiesViewModel.cs:178]: ...which is defined were for non-surface techs?

Comment (daren-thomas) [reply src/Hive.IO/Hive.IO/Forms/ConversionTechPropertiesViewModel.cs:178]: `private static Dictionary<string, ConversionTechDefaults> Defaults`

Which is actually populated by Json.Net - reading in the resource file `Hive.IO.EnergySystems.conversion_technology_defaults.json`. See the `public struct ConversionTechDefaults` for the properties it defines. One of which is a `string Description`. We currently just don't fit anything into it. But I guess we _could_. If we knew where to put it...

**Pull request merged**